File: backend/src/main/java/NPJ/Crewer/running/dto/response/RankingInfo.java
package NPJ.Crewer.running.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.Instant;

@Getter
@AllArgsConstructor
public class RankingInfo {
    private Long recordId;
    private Long runnerId;
    private String runnerNickname;
    private double totalDistance;
    private int totalSeconds;
    private Instant createdAt;
    private String distanceCategory;
    private int ranking;

    // RankingResponse를 DTO RankingInfo로 변환하기 위한 정적 팩토리 메서드
    public static RankingInfo from(RankingResponse rankingResponse) {
        return new RankingInfo(
                rankingResponse.getRecordId(),
                rankingResponse.getRunnerId(),
                rankingResponse.getRunnerNickname(),
                rankingResponse.getTotalDistance(),
                rankingResponse.getTotalSeconds(),
                rankingResponse.getCreatedAt(),
                rankingResponse.getDistanceCategory(),
                rankingResponse.getRanking()
        );
    }
}

File: frontend/lib/models/ranking_info.dart
class RankingInfo {
  final int recordId;
  final int runnerId;
  final String runnerNickname;
  final double totalDistance;
  final int totalSeconds;
  final DateTime createdAt;
  final String distanceCategory;
  final int ranking;

  RankingInfo({
    required this.recordId,
    required this.runnerId,
    required this.runnerNickname,
    required this.totalDistance,
    required this.totalSeconds,
    required this.createdAt,
    required this.distanceCategory,
    required this.ranking,
  });

  // JSON Map을 RankingInfo 객체로 변환하는 팩토리 생성자
  factory RankingInfo.fromJson(Map<String, dynamic> json) {
    return RankingInfo(
      recordId: json['recordId'],
      runnerId: json['runnerId'],
      runnerNickname: json['runnerNickname'],
      totalDistance: (json['totalDistance'] as num).toDouble(),
      totalSeconds: json['totalSeconds'],
      createdAt: DateTime.parse(json['createdAt']),
      distanceCategory: json['distanceCategory'],
      ranking: json['ranking'],
    );
  }
}